flink-core/src/test/java/org/apache/flink/util/MethodForwardingTestUtil.java

```
@@ -0,0 +1,146 @@
+		Preconditions.checkNotNull(skipMethodSet);
+
+		D delegate = spy(delegateClass);
+		W wrapper = wrapperFactory.apply(delegate);
+
+		// ensure that wrapper is a subtype of delegate
+		Preconditions.checkArgument(delegateClass.isAssignableFrom(wrapper.getClass()));
+
+		for (Method delegateMethod : delegateClass.getMethods()) {
+
+			if (checkSkipMethodForwardCheck(delegateMethod, skipMethodSet)) {
+				continue;
+			}
+
+			try {
+				// find the correct method to substitute the bridge for erased generic types.
+				// if this doesn't work, the user need to exclude the method and write an additional test.
+				Method wrapperMethod = wrapper.getClass().getDeclaredMethod(
+					delegateMethod.getName(),
+					delegateMethod.getParameterTypes());
+
+				// things get a bit fuzzy here, best effort to find a match but this might end up with a wrong method.
+				if (wrapperMethod.isBridge()) {
+					for (Method method : wrapper.getClass().getDeclaredMethods()) {
+						if (!method.isBridge()
+							&& method.getName().equals(wrapperMethod.getName())
+							&& method.getParameterCount() == wrapperMethod.getParameterCount()) {
+							wrapperMethod = method;
+							break;
+						}
+					}
+				}
+
+				Class<?>[] parameterTypes = wrapperMethod.getParameterTypes();
+				Object[] arguments = new Object[parameterTypes.length];
+				for (int j = 0; j < arguments.length; j++) {
+					Class<?> parameterType = parameterTypes[j];
+					if (parameterType.isArray()) {
+						arguments[j] = Array.newInstance(parameterType.getComponentType(), 0);
+					} else if (parameterType.isPrimitive()) {
+						arguments[j] = 0;
+					} else {
+						arguments[j] = Mockito.mock(parameterType);
+					}
+				}
+
+				wrapperMethod.invoke(wrapper, arguments);
+				delegateMethod.invoke(Mockito.verify(delegate, Mockito.times(1)), arguments);
+				reset(delegate);
+			} catch (Exception ex) {
+				ex.printStackTrace();
+				fail("Forwarding test failed: " + ex.getMessage());
```

I think we should let the exception propagate.

---

👍